determined-ai/determined
# feat: Add CodeEditor to UI Kit [WEB-1049]

## Description

Main repo + reworking of #6436 

Moves our CodeViewer component to the UI Kit. Changes:

- UI Kit includes examples of Python, YAML, and multiple files displaying a file tree
- files prop (for content) and readonly prop (so it's renamed CodeEditor)
- the previous component handled experimentId internally; here we load the config and file tree in an ExperimentCodeViewer adapter component, and pass the final file list to the component
- the previous component used useSettings to remember the last file viewed by experimentId. Now persistence moves to the page (ExperimentSingleTrialTabs or ExperimentMultiTrialTabs)
- the previous component would alternate between tree-only, code-only, and split views; this was based on user input and window width; for simplicity we now have code-only for one file, split for multiple files.

## Test Plan

On `/det/design`:

- Confirm first CodeEditor is editable, syntax-highlighted Python
- Confirm second is not-editable, syntax-highlighted YAML. There is a working download button and Unicode content of file is intact.
- Confirm third has a file tree with two files and none currently selected. Once a file is selected, the download button picks that file.

On a single-trial experiment (`/det/experiments/1618/code` if you're on latest-master db)
- Confirm the Code tab shows a list of experiment files
- Selecting a config file from the tree displays its YAML; download button works
- Selecting a non-config file from the tree displays its contents; download button works
- Selecting a IPYNB file (available in 1618) displays read-only view using our existing library for this
- Refreshing page persists the last selected file

On a multi-trial experiment which has files in a subfolder (`/det/experiments/2180/code`)
- Confirm the Code tab shows a list of experiment files, including a folder with a file inside
- Selecting the folder's sub-file works and it is downloadable

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
